libgalois/include/katana/PropertyGraph.h

```diff
@@ -122,30 +144,33 @@ class KATANA_EXPORT GraphTopology {
   bool empty() const { return num_nodes() == 0; }
 
 private:
-  // TODO: generalize to typename T
-  static std::shared_ptr<arrow::UInt64Array> LargeToArrowArray(
-      LargeArray<uint64_t>& lg_arr) noexcept {
-    return std::make_shared<arrow::UInt64Array>(
-        lg_arr.size(),
-        arrow::MutableBuffer::Wrap(lg_arr.data(), lg_arr.size()));
-  }
+  friend class GraphTopologyAccess;
 
-  static std::shared_ptr<arrow::UInt32Array> LargeToArrowArray(
-      LargeArray<uint32_t>& lg_arr) noexcept {
-    return std::make_shared<arrow::UInt32Array>(
-        lg_arr.size(),
-        arrow::MutableBuffer::Wrap(lg_arr.data(), lg_arr.size()));
-  }
+  Edge* adj_data() noexcept { return &adj_indices_[0]; }
+  Node* dest_data() noexcept { return &dests_[0]; }
```

Comment:
Ditto about low-level functions. Provide direct access to the member variables and getters is better in my opinion. This seems too specific.